float32 zs_x
float32 zx_x
float32 ys_x
float32 yx_x
float32 o_x
float32 o_y
float32 zs_y
float32 zx_y
float32 ys_y
float32 yx_y
float32 square_num
